TelepresenceBot/server-unit/core/server.js

```diff
@@ -55,10 +57,40 @@ io.sockets.on('connection', function (client) {
 
 
     client.on('disconnect', function() {
+        disconnectRoom(room);
+
         humans.splice(humans.indexOf(client.id), 1);
         bots.splice(bots.indexOf(client.id), 1);
         testClient.emit('disconnected_human', humans);
         testClient.emit('disconnected_bot', bots);
     });
 
+    client.on('move_in', function(direction) {
+        var rooms = Object.keys(io.sockets.adapter.sids[client.id]);
+        for(var i = 0; i < rooms.length; i++) {
+            io.to(rooms[i]).emit('direction', direction);
+            testClient.emit('direction', direction);
+        }
+    });
+
+    function leaveAllRooms(client) {
+        var rooms = Object.keys(io.sockets.adapter.sids[client.id]);
+        for(var i = 0; i < rooms.length; i++) {
+            client.leave(rooms[i]);
+        }
+    }
+
+    function disconnectRoom(name) {
```

Comment:
When one connection drops, either the bot or human, then all other connections in that room are also disconnected.